src/core/control/CrashHandler.cpp

```diff
@@ -1,27 +1,83 @@
 #include "CrashHandler.h"
 
-#include <string>  // for string
+#include <array>        // for std::array
+#include <atomic>       // for std::atomic
+#include <csignal>      // for std::signal
+#include <fstream>      // for std::ofstream
+#include <string>       // for std::string
+#include <string_view>  // for std::string_view
 
-#include <glib.h>  // for g_warning, g_error
+#include <fbbe/stacktrace.h>  // for fbbe::stacktrace
+#include <glib.h>             // for g_warning, g_error
 
 #include "control/xojfile/SaveHandler.h"  // for SaveHandler
-#include "util/PathUtil.h"                // for getConfigFile
-#include "util/i18n.h"                    // for FC, _F, _
+#include "util/PathUtil.h"
+#include "util/PathUtil.h"    // for getConfigFile
+#include "util/Stacktrace.h"  // for Stacktrace::printStracktrace
+#include "util/i18n.h"        // for FC, _F, _
 
+#include "config-dev.h"
 #include "filesystem.h"  // for path
 
-static Document* document = nullptr;
+
+static std::atomic<Document*> document = nullptr;
+static std::atomic<bool> alreadyCrashed = false;
 
 void setEmergencyDocument(Document* doc) { document = doc; }
 
-#ifdef _WIN32
-#include "CrashHandlerWindows.h"
-#else
-#include "CrashHandlerUnix.h"
-#endif
+/**
+ * Print crash log to config directory
+ */
+static void crashHandler(int sig) {
+    if (alreadyCrashed) {  // crasehd again on emergency save
+        exit(2);
+    }
+    alreadyCrashed = true;
+
+    g_warning("[Crash Handler] Crashed with signal %i", sig);
+
+    auto curtime = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+
+    std::array<char, 128> stime;
+    size_t size = 0;
+    {
+        static constexpr std::array<char, 10> error_log_name{"errorlog."};
+        std::memcpy(stime.data(), error_log_name.data(), error_log_name.size() - 1);
+        assert(stime[8] == '.');
+        size = strftime(&stime[9], stime.size() - error_log_name.size(), "%Y%m%d-%H%M%S", localtime(&curtime));
+        std::memcpy(stime.data() + size + 9, ".log", 4);
+    }
```

Comment:
The reason is, that the maximum size is actually determinable, and that we can reduce heap allocations in the signal handler, also I don't know, if IO operations on a string stream are async signal safe. But beside that the handler itself can't be safe, since it writes the error log and does the emergency safe. So probably it doesn't matter. And a cleaner code might be more worth than signal safe and more efficient code. With c++23 we will get spanstream which is what we actually want here.